apps/crn-server/src/data-providers/contentful/tutorial.data-provider.ts

```diff
@@ -37,8 +42,47 @@ type AuthorItem = NonNullable<
 export class TutorialContentfulDataProvider implements TutorialDataProvider {
   constructor(private contentfulClient: GraphQLClient) {}
 
-  async fetch(): Promise<ListResponse<TutorialsDataObject>> {
-    throw new Error('Method not implemented.');
+  async fetch(
+    options: FetchOptions<string[]>,
+  ): Promise<ListResponse<TutorialsDataObject>> {
+    const { take = 10, skip = 0, search } = options;
+
+    const where: TutorialsFilter = {};
+
+    const searchTerms = (search || '').split(' ').filter(Boolean);
+
+    if (searchTerms.length) {
+      where.OR = [
+        ...searchTerms.map((term) => ({ title_contains: term })),
+        ...searchTerms.map((term) => ({ shortText_contains: term })),
+        ...searchTerms.map((term) => ({ tags: { name: term } })),
+        ...searchTerms.map((term) => ({ teams: { displayName: term } })),
+      ];
+    }
+    const { discoverCollection } = await this.contentfulClient.request<
```

Comment:
```suggestion
    const { tutorialCollection } = await this.contentfulClient.request<
```

Reply:
We pull tutorials to the frontend through the trainings field under the discover model, which is why the query fetches discoverCollection instead of tutorialsCollection